x-pack/plugins/reporting/server/routes/diagnose/config.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import os from 'os';
+import numeral from '@elastic/numeral';
+import { defaults, get } from 'lodash';
+import { ReportingCore } from '../..';
+import { API_DIAGNOSE_URL } from '../../../common/constants';
+import { authorizedUserPreRoutingFactory } from '../lib/authorized_user_pre_routing';
+import { LevelLogger as Logger } from '../../lib';
+
+const KIBANA_MAX_SIZE_BYTES_PATH = 'csv.maxSizeBytes';
+const ES_MAX_SIZE_BYTES_PATH = 'http.max_content_length';
+const TWO_GIGABYTES = 2147483648; // ONE POINT TWENTY ONE JIGAWATTS?!?
+
+interface ConfigResponse {
+  help: string[];
+  success: boolean;
+  logs: string;
+}
+
+export const registerDiagnoseConfig = (reporting: ReportingCore, logger: Logger) => {
+  const setupDeps = reporting.getPluginSetupDeps();
+  const userHandler = authorizedUserPreRoutingFactory(reporting);
+  const { router, elasticsearch } = setupDeps;
+
+  router.post(
+    {
+      path: `${API_DIAGNOSE_URL}/config`,
+      // Currently no supported params, but keeping it open
+      // in case the need/want arises
+      validate: {},
+    },
+    userHandler(async (user, context, req, res) => {
+      const warnings = [];
+      const { callAsInternalUser } = elasticsearch.legacy.client;
+      const config = reporting.getConfig();
+      const memAvailable = os.totalmem();
```

Comment:
I think this value will give the wrong value when we're running in a container, and there is a memory limit on the container. We would want it to tell us what the container's memory limit is, but I'm pretty sure this just gets the host memory.

This article: https://developer.ibm.com/technologies/node-js/articles/nodejs-memory-management-in-container-environments/ talks about how to check the memory limits in a container

Reply:
Yup, removed until we kind find a better value.

FWIW platform does this same exact call in their setup (os. totalmem), we might want to alert them to that...